VTX LED Strip

sheaivey opened the pull request:
This PR goes in tandem with Betaflight PR [#3250](https://github.com/betaflight/betaflight/pull/3250)

This PR is in regards to [Change RGB LED color based on VTX channel](https://github.com/betaflight/betaflight/issues/3228)

<img width="830" alt="screen shot" src="https://user-images.githubusercontent.com/3945263/27006923-a0e881ae-4df6-11e7-9ba3-5a3ca37614fc.png">

I wound up replacing the Blink On Landing overlay in order to reduce the amount of changes needed to keep the firmware in sync with he configurator.

*I apologize for the whitespace cleanup.*

mikeller commented on tabs/led_strip.html:
```diff
@@ -123,11 +115,15 @@
                 </div>
+                <div class="checkbox vtxOverlay">
+                    <input type="checkbox" name="Vtx" class="toggle function-v" />
+                    <label> <span>VTX (uses vtx frequency to assign color)</span></label>
```
Can you please put GUI text into https://github.com/betaflight/betaflight-configurator/blob/master/_locales/en/messages.json , to make translation easier? They then can be used like this: https://github.com/sheaivey/betaflight-configurator/blob/dbd2fdace67858430f71f06b64d7b2ef16b1d052/tabs/led_strip.html#L3

mikeller commented on tabs/led_strip.js:
```diff
@@ -64,26 +64,26 @@ TABS.led_strip.initialize = function (callback, scrollPosition) {
         for (var i = 0; i < 256; i++) {
             if (semver.lte(CONFIG.apiVersion, "1.19.0"))
-                theHTML[theHTMLlength++] = ('<div class="gPoint"><div class="indicators"><span class="north"></span><span class="south"></span><span class="west"></span><span class="east"></span><span class="up">U</span><span class="down">D</span></div><span class="wire"></span><span class="overlay-t"> </span><span class="overlay-s"> </span><span class="overlay-w"> </span><span class="overlay-i"> ...
+                theHTML[theHTMLlength++] = ('<div class="gPoint"><div class="indicators"><span class="north"></span><span class="south"></span><span class="west"></span><span class="east"></span><span class="up">U</span><span class="down">D</span></div><span class="wire"></span><span class="overlay-t"> </span><span class="overlay-s"> </span><span class="overlay-w"> </span><span class="overlay-v"> ...
```
This shouldn't be changed, since this change in the firmware is not in firmware that is apiVersion 1.19 or older.

mikeller commented on tabs/led_strip.js:
```diff
@@ -64,26 +64,26 @@ TABS.led_strip.initialize = function (callback, scrollPosition) {
+                theHTML[theHTMLlength++] = ('<div class="gPoint"><div class="indicators"><span class="north"></span><span class="south"></span><span class="west"></span><span class="east"></span><span class="up">U</span><span class="down">D</span></div><span class="wire"></span><span class="overlay-t"> </span><span class="overlay-s"> </span><span class="overlay-w"> </span><span class="overlay-v"> ...
             else
-                theHTML[theHTMLlength++] = ('<div class="gPoint"><div class="indicators"><span class="north"></span><span class="south"></span><span class="west"></span><span class="east"></span><span class="up">U</span><span class="down">D</span></div><span class="wire"></span><span class="overlay-t"> </span><span class="overlay-o"> </span><span class="overlay-b"> </span><span class="overlay-n"> ...
+                theHTML[theHTMLlength++] = ('<div class="gPoint"><div class="indicators"><span class="north"></span><span class="south"></span><span class="west"></span><span class="east"></span><span class="up">U</span><span class="down">D</span></div><span class="wire"></span><span class="overlay-t"> </span><span class="overlay-o"> </span><span class="overlay-b"> </span><span class="overlay-n"> ...
```
Can you make the changed html conditional on apiVersion being >= 1.36? This way, the configurator will still be correct for users with older firmware.

mikeller commented on tabs/led_strip.js:
```diff
@@ -766,51 +771,71 @@ TABS.led_strip.initialize = function (callback, scrollPosition) {
+    }
+
+    function isVtxActive(activeFunction) {
+        switch (activeFunction) {
```
Another apiVersion switch needed here.

mikeller commented on tabs/led_strip.html:
```diff
@@ -54,11 +54,11 @@
             <button class="funcClearAll">Clear ALL</button>
 
-            <div class="section">LED Functions</div>
+            <div class="section" i18n="ledStripFunctionSection">LED Functions</div>
```
Thanks for pulling out the UI strings.

I've never tried what happens if you leave the text content of the element in when you use `i18n` (presumably the text is overwritten), but I think it's best to remove the text content, to stop people from updating it (instead of the text in `messages.json`) when they change things.

sheaivey replied:
Just saw this I'll get it fixed now.

sheaivey commented:
Thanks for the review @mikeller. I added checks around all the added VTX Overlay code. I tested with older firmwares and confirmed that Blink While Landing shows up in apiVersions before 1.36.0. After 1.36.0 the VTX Overlay option is shown and blink on landing is hidden.

I'll do better in the future to not commit whitespace changes. The review tool I use ignores white space otherwise I might have caught that before committing.

I went ahead and rebased and squashed the commits. Let me know if there is any other changes needed.
